core/src/main/java/io/github/fourlastor/game/coordinates/Hex.java

```diff
@@ -65,4 +64,57 @@ public void set(Hex other) {
     public boolean sameAxisAs(Hex hex) {
         return hex.cube.x == cube.x || hex.cube.y == cube.y || hex.cube.z == cube.z;
     }
+
+    public Hex offset(Direction direction, int amount) {
+        return new Hex(cube.cpy().add(direction.cube.cpy().scl(amount)));
+    }
+
+    public enum Direction {
+        S(0, -1, 1),
+        SE(1, -1, 0),
+        NE(1, 0, -1),
+        N(0, +1, -1),
+        NW(-1, 1, 0),
+        SW(-1, 0, 1);
+
+        private final GridPoint3 cube;
+
+        Direction(int x, int y, int z) {
+            cube = new GridPoint3(x, y, z);
+        }
+
+        public static Direction byValue(GridPoint3 cube) {
+            return Arrays.stream(values())
+                    .filter(dir -> dir.cube.equals(cube))
+                    .findFirst()
+                    .get();
+        }
+
+        public static Direction byName(String value) {
+
+            Optional<Direction> test = Arrays.stream(values())
+                    .filter(dir -> dir.name().equalsIgnoreCase(value))
+                    .findFirst();
+            return test.orElse(null);
+        }
+
+        public Direction opposite() {
+            return byValue(cube.cpy().inverse());
+        }
+
+        /**
+         * Starting at 0 degrees = East direction, counter-clockwise.
+         */
+        public static Direction fromRotation(int degrees) {
+
+            degrees = Math.floorMod(degrees, 360);
+            if (degrees <= 60) return NE;
+            else if (degrees <= 120) return N;
+            else if (degrees <= 180) return NW;
+            else if (degrees <= 240) return SW;
+            else if (degrees <= 300) return S;
+            else if (degrees <= 360) return SE;
+            return null;
```

Comment:
I'd change this to `throw new IllegalArgumentException("No direction found for rotation " + degrees);`

Reply:
I think it would also be ok to remove the last if and just have `else return SE` (as we're ensured of that by the `floorMod` above)